docs/csharp/language-reference/keywords/volatile.md

```diff
@@ -9,42 +9,46 @@ helpviewer_keywords:
   
--   An enum type with one of the following base types: byte, sbyte, short, ushort, int, or uint.  
+- Simple types such as `sbyte`, `byte`, `short`, `ushort`, `int`, `uint`, `char`, `float`, and `bool`.  
   
--   Generic type parameters known to be reference types.  
+- An `enum` type with one of the following base types: `byte`, `sbyte`, `short`, `ushort`, `int`, or `uint`.  
   
--   <xref:System.IntPtr> and <xref:System.UIntPtr>.  
+- Generic type parameters known to be reference types. 
   
- The volatile keyword can only be applied to fields of a class or struct. Local variables cannot be declared `volatile`.  
+- <xref:System.IntPtr> and <xref:System.UIntPtr>.  
   
-## Example  
- The following example shows how to declare a public field variable as `volatile`.  
+The volatile keyword can only be applied to fields of a `class` or `struct`. Local variables cannot be declared `volatile`.
   
- [!code-csharp[csrefKeywordsModifiers#24](../../../csharp/language-reference/keywords/codesnippet/CSharp/volatile_1.cs)]  
-  
-## Example  
- The following example demonstrates how an auxiliary or worker thread can be created and used to perform processing in parallel with that of the primary thread. For background information about multithreading, see [Managed Threading](../../../standard/threading/index.md) and [Threading (C#)](../../programming-guide/concepts/threading/index.md).  
+## Example
+
+The following example shows how to declare a public field variable as `volatile`.  
   
- [!code-csharp[csProgGuideThreading#1](../../../csharp/language-reference/keywords/codesnippet/CSharp/volatile_2.cs)]  
+[!code-csharp[declareVolatile](~/samples/snippets/csharp/language-reference/keywords/volatile/Program.cs#Declaration)]
+
+The following example demonstrates how an auxiliary or worker thread can be created and used to perform processing in parallel with that of the primary thread. For background information about multithreading, see [Managed Threading](../../../standard/threading/index.md) and [Threading (C#)](../../programming-guide/concepts/threading/index.md).  
   
-## C# Language Specification  
- [!INCLUDE[CSharplangspec](~/includes/csharplangspec-md.md)]  
+[!code-csharp[declareVolatile](~/samples/snippets/csharp/language-reference/keywords/volatile/Program.cs#Volatile)]
+
+With the `volatile` modifier on  the declaration of  in place, you'll always get the same results (similar to the excerpt shown in the preceding code). However, without that modifier on the `_shouldStop` member, the behavior is unpredictable. The `DoWork` method may optimize the member access, resulting in reading stale data. Because of the nature of multi-threaded programming, the number of stale reads is unpredictable. Different runs of the program will produce somewhat different results.
+
+## C# Language Specification
+
+[!INCLUDE[CSharplangspec](~/includes/csharplangspec-md.md)]  
```

Comment:
To follow #7890, please consider to give the following link:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/classes#volatile-fields